doc/regexp.rdoc

```diff
@@ -1,753 +1,1176 @@
-is ASCII-compatible.
+    # Same as Regexp.new('FOO', Regexp::NOENCODING)
+    /FOO/n.encoding # => #<Encoding:US-ASCII>
 
-If a match between incompatible encodings is attempted an
-<tt>Encoding::CompatibilityError</tt> exception is raised.
+- <tt>/_pat_/u</tt>: UTF-8
 
-The <tt>Regexp#fixed_encoding?</tt> predicate indicates whether the regexp
-has a <i>fixed</i> encoding, that is one incompatible with ASCII. A
-regexp's encoding can be explicitly fixed by supplying
-<tt>Regexp::FIXEDENCODING</tt> as the second argument of
-<tt>Regexp.new</tt>:
+    /foo/u.encoding # => #<Encoding:UTF-8>
 
-    r = Regexp.new("a".force_encoding("iso-8859-1"),Regexp::FIXEDENCODING)
-    r =~ "a\u3042"
-       # raises Encoding::CompatibilityError: incompatible encoding regexp match
-       #         (ISO-8859-1 regexp with UTF-8 string)
+- <tt>/_pat_/e</tt>: EUC-JP
 
-== \Regexp Global Variables
+    /foo/e.encoding # => #<Encoding:EUC-JP>
 
-Pattern matching sets some global variables :
+- <tt>/_pat_/s</tt>: Windows-31J
 
-* <tt>$~</tt> is equivalent to Regexp.last_match;
-* <tt>$&</tt> contains the complete matched text;
-* <tt>$`</tt> contains string before match;
-* <tt>$'</tt> contains string after match;
-* <tt>$1</tt>, <tt>$2</tt> and so on contain text matching first, second, etc
-  capture group;
-* <tt>$+</tt> contains last capture group.
+    /foo/s.encoding # => #<Encoding:Windows-31J>
 
-Example:
+A regexp can be matched against a target string when either:
 
-    m = /s(\w{2}).*(c)/.match('haystack') #=> #<MatchData "stac" 1:"ta" 2:"c">
-    $~                                    #=> #<MatchData "stac" 1:"ta" 2:"c">
-    Regexp.last_match                     #=> #<MatchData "stac" 1:"ta" 2:"c">
-
-    $&      #=> "stac"
-            # same as m[0]
-    $`      #=> "hay"
-            # same as m.pre_match
-    $'      #=> "k"
-            # same as m.post_match
-    $1      #=> "ta"
-            # same as m[1]
-    $2      #=> "c"
-            # same as m[2]
-    $3      #=> nil
-            # no third group in pattern
-    $+      #=> "c"
-            # same as m[-1]
-
-These global variables are thread-local and method-local variables.
+- They have the same encoding.
+- The regexp's encoding is _US-ASCII_ and the string's encoding
+  is ASCII-compatible.
+
+If a match between incompatible encodings is attempted an
+<tt>Encoding::CompatibilityError</tt> exception is raised.
+
+Method Regexp#fixed_encoding?returns whether the regexp
+has a <i>fixed</i> encoding: one that is incompatible with ASCII.
+
+The encoding may be explicitly fixed by including Regexp::FIXEDENCODING
+in the second argument for Regexp.new:
+
+  # Regexp with encoding ISO-8859-1.
+  re = Regexp.new("a".force_encoding('iso-8859-1'), Regexp::FIXEDENCODING)
+  re.encoding  # => #<Encoding:ISO-8859-1>
+  # Target string with encoding UTF-8.
+  s = "a\u3042"
+  s.encoding   # => #<Encoding:UTF-8>
+  re.match(s)  # Raises Encoding::CompatibilityError.
 
 == Performance
```

Comment:
I recommend removing this section as it does not reflect optimizations made in Ruby 3.2.

Reply:
Removed.